### PR TITLE
Fixed pom versioning issue as part of #277

### PR DESCRIPTION
**Pull request by fabmars**
Fixed pom versioning issue as part of #277
Only thing is, I'm not even letting a default in <revision> else it stays on the resulting pom.xml (in <revision/>, not in <version/>, duh), but I don't thing it's a problem.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -27,8 +27,6 @@
   <version>${revision}</version>
 
   <properties>
-    <revision>2.2-SNAPSHOT</revision>
```
Comment (kwhat) [inline pom.xml:30]: Why are we removing revision here but leaving the ref `<version>` above?

File: pom.xml [hunk 2]
```diff
@@ -140,6 +138,33 @@
             </executions>
           </plugin>
 
+          <plugin>
```
Comment (kwhat) [inline pom.xml:141]: I have a feeling this should be closer to line 327 instead of in the jdk 8 profile.

**Review by kwhat — COMMENTED** [review]
I just started playing with this and I still have no idea how it works.

File: pom.xml [hunk 3]
```diff
@@ -140,6 +138,33 @@
...
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>flatten-maven-plugin</artifactId>
+            <version>1.2.2</version>
+            <configuration>
+              <updatePomFile>true</updatePomFile>
```
Comment (kwhat) [inline pom.xml:146]: Lets add `<outputDirectory>${project.build.directory}</outputDirectory>` above `updatePomFile>`

File: pom.xml [hunk 4]
```diff
@@ -140,6 +138,33 @@
             </executions>
           </plugin>
 
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>flatten-maven-plugin</artifactId>
+            <version>1.2.2</version>
+            <configuration>
```
Comment (kwhat) [inline pom.xml:145]: blank line between `<version>` & `<configuration>`

File: pom.xml [hunk 5]
```diff
@@ -140,6 +138,33 @@
...
+              <updatePomFile>true</updatePomFile>
+              <flattenMode>resolveCiFriendliesOnly</flattenMode>
+            </configuration>
+            <executions>
```
Comment (kwhat) [inline pom.xml:149]: blank line between `</configuration>` & `<executions>`

File: pom.xml [hunk 6]
```diff
@@ -140,6 +138,33 @@
...
+              <execution>
+                <id>flatten</id>
+                <phase>process-resources</phase>
+                <goals>
+                    <goal>flatten</goal>
+                </goals>
+              </execution>
+              <execution>
```
Comment (kwhat) [inline pom.xml:157]: blank line between `</execution>` & `<execution>`

File: pom.xml [hunk 7]
```diff
@@ -140,6 +138,33 @@
...
+                <id>flatten.clean</id>
+                <phase>clean</phase>
+                <goals>
+                    <goal>clean</goal>
+                </goals>
+              </execution>
+            </executions>
+            <inherited>true</inherited>
```
Comment (kwhat) [inline pom.xml:165]: blank line between `</executions>` & `<inherited>`

Comment (kwhat) [reply pom.xml:165]: Is `<inherited>true</inherited>` required?

**Review by kwhat — CHANGES_REQUESTED** [review]

**Comment by kwhat** [comment]
I think it is working.  I tested a few things and it appears to be creating the correct pom now.

File: pom.xml (hunk 1 above)
Comment (kwhat) [reply pom.xml:30]: I think we can leave this in as the default so you don't have to build with `-Drevision` every time.

**Comment by kwhat** [comment]
I just pulled it in with a few other maven fixes.

File: pom.xml (hunk 1 above)
Comment (fabmars) [reply pom.xml:30]: I agree leaving it lifts the requirement to specify it, but the resulting pom will still contain the <revision> line with the hardcoded snapshot version on top the <version> line with the correct interpolation. Might be misleading (for whoever opens the jar and checks the pom :)).

Comment (kwhat) [reply pom.xml:30]: I don't think this is the case for the release build.  Ex: `mvn -Drevision="2.2.1" package` produces a jnativehook-2.2.1.jar  file with following in the pom:

```
<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>com.github.kwhat</groupId>
  <artifactId>jnativehook</artifactId>
  <version>2.2.1</version>
  <name>JNativeHook</name>
  <description>Global keyboard and mouse listeners for Java.</description>
  <url>https://github.com/kwhat/jnativehook</url>
```

It still has the following, but I don't think that would be a problem would it?

```
  <properties>
    <maven.compiler.debug>true</maven.compiler.debug>
    <maven.compiler.multi-release>false</maven.compiler.multi-release>
    <maven.compiler.compilerId>javac</maven.compiler.compilerId>
    <maven.compiler.debuglevel>lines,vars,source</maven.compiler.debuglevel>
    <revision>2.2-SNAPSHOT</revision>
```

What was this new plugin suppose to resolve?

Comment (fabmars) [reply pom.xml:30]: I don't think that would be a problem.
and the plugin is just solving the <version>${revision}</version> interpolation issue.

**Pull request closed without merging**